torch/profiler/profiler.py

```diff
@@ -46,6 +46,95 @@ def supported_activities():
     return torch.autograd._supported_activities()
 
 
+class ExecutionTraceObserver:
```

Comment:
This class was moved up so we can refer to it in other classes

Reply:
What do you think about abstracting out the observer to an interface `ITraceObserver` and then provide this `ExecutionTraceObserver` as an implementation of that interface for the community? The functions are very generic anyways (register/unregister/start/stop etc).

Reply:
done, please see update